src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -256,6 +569,133 @@ class GeometricInt64Generator : public Generator<int64_t> {
     const double _p;
 };
 
+// This generator allows choosing any valid generator, incuding documents. As such it cannot be used
```

Comment:
I'm open to ideas here. This allows be to choose anything, which is cool, but without a `std::string evaluate()` method, it can't be called within the JoinGenerator. I made a specific string only version of this that will work with join, but it would be nice to have it all work in one.